### PR TITLE
Test TAILs on sources with empty frontiers

### PR DESCRIPTION
**Pull request by JLDLaughlin**
Addresses the testing concerns of this issue: (https://github.com/MaterializeInc/materialize/issues/3468).

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3500)

### CODE REVIEW

File: test/testdrive/sinks.td [hunk 1]
```diff
@@ -116,3 +116,29 @@ $ kafka-verify format=avro sink=materialize.public.snk8
 {"before": null, "after": {"a": "jack", "b": "jill", "mz_line_no": 2}}
 {"before": null, "after": {"a": "goofus", "b": "gallant", "mz_line_no": 3}}
 {"before": null, "after": {"a": "extra", "b": "row", "mz_line_no": 4}}
+
+# Test that we are correctly handling WITH/WITHOUT SNAPSHOT.
+> CREATE MATERIALIZED VIEW foo AS VALUES (1), (2), (3);
+
+# Make sure data is read into view before we TAIL it.
```
Comment (ruchirK) [inline test/testdrive/sinks.td:123]: I'm surprised that this bit was necessary. `avro-sinks.td` creates a static view and immediately `SINK`s it and thats been fine afaik. if thats wrong then I can amend those tests

File: src/materialized/tests/sql.rs [hunk 2]
```diff
@@ -211,6 +211,35 @@ fn test_tail() -> Result<(), Box<dyn Error>> {
...
+    // All records should be read into view before we start tailing.
+    thread::sleep(Duration::from_millis(100));
+
+    let mut tail_reader = client.copy_out("TAIL foo WITHOUT SNAPSHOT")?.split(b'\n');
+    let mut without_snapshot_count = 0;
+    while let Some(_value) = tail_reader.next().transpose()? {
+        without_snapshot_count += 1;
+    }
+    assert_eq!(0, without_snapshot_count);
+    drop(tail_reader);
```
Comment (ruchirK) [inline src/materialized/tests/sql.rs:231]: why do we need to call `drop` here explicitly? asking because I haven't seen that in our codebase before

Comment (JLDLaughlin) [reply src/materialized/tests/sql.rs:231]: the `client.copy_out` is a mutable borrow of the client, so without the explicit `drop` of the `tail_reader` I am unable to reuse the same client on line 233!

File: test/testdrive/sinks.td (hunk 1 above)
Comment (JLDLaughlin) [reply test/testdrive/sinks.td:123]: it is working without the `SELECT`!

**Pull request merged**